bluesky.py

```diff
@@ -0,0 +1,90 @@
+import models
+from granary import bluesky as gr_bluesky
+from oauth_dropins import bluesky as oauth_bluesky
+from flask_app import app
+import util
+import logging
+from flask import flash, render_template
+from oauth_dropins.webutil.util import json_loads
+from urllib.parse import quote
+
+logger = logging.getLogger(__name__)
+
+
+class Bluesky(models.Source):
+  """
+  A Bluesky account.
+  """
+  SHORT_NAME = 'bluesky'
+  GR_CLASS = gr_bluesky.Bluesky
+  OAUTH_START = oauth_bluesky.Start
+  AUTH_MODEL = oauth_bluesky.BlueskyAuth
+  URL_CANONICALIZER = util.UrlCanonicalizer(
+          domain=GR_CLASS.DOMAIN,
+          # Bluesky does not support HEAD requests.
+          redirects=False)
+
+  @staticmethod
+  def new(auth_entity, **kwargs):
+    """Creates and returns a :class:`Bluesky` entity.
+
+    Args:
+      auth_entity: :class:`oauth_bluesky.BlueskyAuth`
+      kwargs: property values
+    """
+    assert 'username' not in kwargs
+    assert 'id' not in kwargs
+    user = json_loads(auth_entity.user_json)
+    (_, app_password) = auth_entity.access_token()
+    gr_source = gr_bluesky.Bluesky(user.get('handle'), app_password=app_password)
+    actor = gr_source.user_to_actor(user)
+    return Bluesky(id=auth_entity.key_id(),
+                   username=auth_entity.key_id(),
+                   auth_entity=auth_entity.key,
+                   name=user.get('handle'),
```

Comment:
and then this should be displayName?

Reply:
It could be but I think that would actually be inconsistent with other silos? Bridgy current seems to use username rather than display name for this in Mastodon and Twitter (RIP):

<img width="469" alt="image" src="https://github.com/snarfed/bridgy/assets/2765356/a4c3ef42-9fad-4f40-97e8-3256f74a5fb5">

Reply:
Hmm, unless this is something that changed recently? Looking at the source code it appears to be as you say, but that isn't reflected in my actual brid.gy account

Reply:
right, Bridgy shows silo-specific usernames/handles in the UI, but those come from `label_name()`, not from the `name` property. I think we want `name` to be the human-readable name.

https://github.com/snarfed/bridgy/blob/33b4a717c0c0686ecc308d15b55836e23801b8e7/templates/profile_link.html#L12

https://github.com/snarfed/bridgy/blob/33b4a717c0c0686ecc308d15b55836e23801b8e7/models.py#L143

Checking in a few other silos...yup, `name` should be human name.

Reply:
OK I'm a bit baffled by this, running it locally gives the same result, but the name field is clearly my display name, it just doesn't get used in the display, but in Bluesky it does!

<img width="540" alt="image" src="https://github.com/snarfed/bridgy/assets/2765356/08a20d0f-39b3-40d8-8fa1-f4e16a769236">

I guess the problem is somewhere else?

Reply:
Ah, so I should change the implementation of `label_name()`?

Reply:
Ahh sorry, the missing piece is that `Bluesky` itself should override `label_name()` with whatever we want there. Other examples:

https://github.com/snarfed/bridgy/blob/main/mastodon.py#L119-L121

https://github.com/snarfed/bridgy/blob/33b4a717c0c0686ecc308d15b55836e23801b8e7/github.py#L74-L76

so yeah I think we want to do that and return handle.

Reply:
Sorry for the confusion!

Reply:
I wish GitHub had auto-refresh 😆 Will get that changed!